### DISCUSSION TITLE
How do I know what a VM Error code means?

### DISCUSSION THREAD

**Discussion by gregnazario** [aptos-labs/aptos-developer-discussions#46] [Questions] [answered]
From lanford33 on Discord

What should I do with this error?
```
body: "{\"status\":400,\"errorCode\":\"vm_error\",\"vmErrorCode\":2012}",
```

**Comment by gregnazario** [comment]
This happens to be a special case where the code comes from the VM, and doesn't have a message (which is odd).  Which network are you running on?  Mainnet?

The code is this:
```
 VM_STARTUP_FAILURE = 2012,
```

https://github.com/aptos-labs/aptos-core/blob/main/third_party/move/move-core/types/src/vm_status.rs#L734

**Reply by LanfordCai** [reply]
Thank you for your response. I just try to transfer some APT from one address to another on the mainnet, why does this error be raised?

**Reply by gregnazario** [reply]
Which wallet are you using?

**Reply by LanfordCai** [reply]
I used the Aptos TS SDK 1.3.16 to build the transaction. The code had been written several months ago, and it worked fine.

**Reply by gregnazario** [reply, marked as answer]
What fullnode do you use?  Or do you use your own?

If you're using your own, you may need to update your software on your fullnode

**Reply by LanfordCai** [reply]
Oh, it works fine after I upgraded my node, thank you!
